### PR TITLE
[server] Remove optional in schema lookup.

### PR DESCRIPTION
**Pull request by majisourav99**
## Summary, imperative, start upper case, don't end with a period

Remove unnecessary optional object creations in schema lookup.

## How was this PR tested?

CI build.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Comment by adamxchen** [comment]
Feel free to bypass the CI check as long as the WC test and integ tests passed.

File: internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java [hunk 1]
```diff
@@ -246,15 +246,14 @@ public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
   }
 
   @Override
-  public Optional<SchemaEntry> getSupersetSchema(String storeName) {
-    Optional<Integer> supersetSchemaID = getSupersetSchemaID(storeName);
-    return supersetSchemaID.map(schemaID -> accessor.getValueSchema(storeName, String.valueOf(schemaID)));
+  public SchemaEntry getSupersetSchema(String storeName) {
+    Integer supersetSchemaID = getSupersetSchemaID(storeName);
```
Comment (adamxchen) [inline internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java:250]: Why don't we  have to check whether the `supersetSchemaID` is valid or not, just like some changes you made above?

**Review by adamxchen — CHANGES_REQUESTED** [review]
Thanks for the cleanup. One comment left. And feel free to bypass the CI code coverage check when integrations tests and other checks pass.

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepository.java [hunk 2]
```diff
@@ -296,31 +286,33 @@ public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
...
+  public SchemaEntry getSupersetSchema(String storeName) {
     fetchStoreSchemaIfNotInCache(storeName);
     SchemaData schemaData = schemaMap.get(storeName);
     if (schemaData == null) {
       throw new VeniceNoStoreException(storeName);
     }
 
-    Optional<Integer> supersetSchemaID = getSupersetSchemaID(storeName);
-    return supersetSchemaID.map(schemaData::getValueSchema);
+    Integer supersetSchemaID = getSupersetSchemaID(storeName);
```
Comment (FelixGV) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepository.java:309]: Same question here

File: internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java [hunk 3]
```diff
@@ -246,15 +246,14 @@ public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
...
+    return accessor.getValueSchema(storeName, String.valueOf(supersetSchemaID));
   }
 
-  private Optional<Integer> getSupersetSchemaID(String storeName) {
+  private Integer getSupersetSchemaID(String storeName) {
```
Comment (FelixGV) [inline internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java:254]: And here

**Review by FelixGV — COMMENTED** [review]
Great initiative! Thanks.

File: internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java (hunk 1 above)
Comment (majisourav99) [reply internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java:250]: Actually I could short the lookup and return null if its invalid.

**Review by FelixGV — DISMISSED** [review]

**Comment by FelixGV** [comment]
Looks like you need to add some unit tests for the modified code

**Review by nisargthakkar — APPROVED** [review]

**Pull request merged**